includes/init_includes/init_customer_auth.php

```diff
@@ -79,27 +79,34 @@
  * 2 = customer authorization pending can browse but no prices
  */
 switch (true) {
+  /**
+   * bypass redirects for these scripts, to processing regardless of store mode or cust auth mode
+   */
+  case (preg_match('/ipn_main_handler\.php$/', $_SERVER['SCRIPT_NAME'])):

```

Comment:
I hadn't considered forcing it to test which directory it was in, or jailing/sandboxing it.

I guess the answer to your question about "desirable" is: it depends on what that "modification" was for in the first place.